llvm/lib/Support/VirtualFileSystem.cpp

```diff
@@ -893,6 +893,10 @@ bool InMemoryFileSystem::addFile(const Twine &P, time_t ModificationTime,
     }
 
     if (auto *NewDir = dyn_cast<detail::InMemoryDirectory>(Node)) {
+      // Trying to insert a file in place of a directory.
```

Comment:
not sure if this is always the case, we might as well be inserting new-file as a directory

Reply:
With the refactored code we cover the case that a directory is added